src/checkbox/checkbox.js

```diff
@@ -15,255 +15,180 @@
+    }
 
-  /**
-   * Store constants in one place so they can be updated easily.
-   *
-   * @enum {string | number}
-   * @private
-   */
-  MaterialCheckbox.prototype.Constant_ = {
-    TINY_TIMEOUT: 0.001
-  };
+    // Initialize event listeners.
+    this.changeListener_ = this.refresh.bind(this);
+    this.focusListener_ =
+        () => this.root_.classList.add(MaterialCheckbox.classes_.IS_FOCUSED);
+    this.blurListener_ =
+        () => this.root_.classList.remove(MaterialCheckbox.classes_.IS_FOCUSED);
+    this.mouseUpListener_ = this.blur_.bind(this);
 
-  /**
-   * Store strings for class names defined by this component that are used in
-   * JavaScript. This allows us to simply change it in one place should we
-   * decide to modify at a later date.
-   *
-   * @enum {string}
-   * @private
-   */
-  MaterialCheckbox.prototype.CssClasses_ = {
-    INPUT: 'mdl-checkbox__input',
-    BOX_OUTLINE: 'mdl-checkbox__box-outline',
-    FOCUS_HELPER: 'mdl-checkbox__focus-helper',
-    TICK_OUTLINE: 'mdl-checkbox__tick-outline',
-    RIPPLE_EFFECT: 'mdl-js-ripple-effect',
-    RIPPLE_IGNORE_EVENTS: 'mdl-js-ripple-effect--ignore-events',
-    RIPPLE_CONTAINER: 'mdl-checkbox__ripple-container',
-    RIPPLE_CENTER: 'mdl-ripple--center',
-    RIPPLE: 'mdl-ripple',
-    IS_FOCUSED: 'is-focused',
-    IS_DISABLED: 'is-disabled',
-    IS_CHECKED: 'is-checked',
-    IS_UPGRADED: 'is-upgraded'
-  };
+    // Finalize initialization.
+    this.init_();
+  }
 
   /**
-   * Handle change of state.
+   * String constants used in this component.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @protected
    */
-  MaterialCheckbox.prototype.onChange_ = function(event) {
-    this.updateClasses_();
-  };
+  static get strings_() {
+    return {
+      CLASS_NAME: 'MaterialCheckbox'
+    };
+  }
 
   /**
-   * Handle focus of element.
+   * CSS classes used in this component.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @protected
    */
-  MaterialCheckbox.prototype.onFocus_ = function(event) {
-    this.element_.classList.add(this.CssClasses_.IS_FOCUSED);
-  };
+  static get classes_() {
+    return {
+      ROOT: 'mdl-checkbox',
+      JS: 'mdl-js-checkbox',
+      INPUT: 'mdl-checkbox__input',
 
-  /**
-   * Handle lost focus of element.
-   *
-   * @param {Event} event The event that fired.
-   * @private
-   */
-  MaterialCheckbox.prototype.onBlur_ = function(event) {
-    this.element_.classList.remove(this.CssClasses_.IS_FOCUSED);
-  };
+      IS_FOCUSED: 'is-focused',
+      IS_DISABLED: 'is-disabled',
+      IS_CHECKED: 'is-checked',
+      IS_UPGRADED: 'is-upgraded'
+    };
+  }
 
   /**
-   * Handle mouseup.
+   * Attach all listeners to the DOM.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @export
    */
-  MaterialCheckbox.prototype.onMouseUp_ = function(event) {
-    this.blur_();
-  };
+  addEventListeners() {
```

Comment:
Yes. There are legitimate use cases for this, such as e.g. implementing modal dialog behaviour or ensuring garbage collection of the event handlers.